yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,418 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/dustin/go-humanize"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/dbzm"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/metadb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var (
+	backupSchemaFiles    utils.BoolStr
+	backupDataFiles      utils.BoolStr
+	saveMigrationReports utils.BoolStr
+	backupLogFiles       utils.BoolStr
+	backupDir            string
+)
+
+var endMigrationCmd = &cobra.Command{
+	Use:   "migration",
+	Short: "End the current migration and cleanup all metadata stored in databases(Target, FF and FB) and export-dir",
+	Long:  "End the current migration and cleanup all metadata stored in databases(Target, FF and FB) and export-dir",
+
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := validateEndMigrationFlags(cmd)
+		if err != nil {
+			utils.ErrExit(err.Error())
+		}
+
+		if utils.IsDirectoryEmpty(exportDir) {
+			utils.ErrExit("export directory is empty, nothing to end")
+		}
+	},
+
+	Run: endMigrationCommandFn,
+}
+
+func endMigrationCommandFn(cmd *cobra.Command, args []string) {
+	if utils.AskPrompt("Migration can't be resumed or continued after this.", "Are you sure you want to end the migration") {
+		log.Info("ending the migration")
+	} else {
+		log.Info("Aborting the end migration command")
+		return
+	}
+
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("end migration: getting migration status record: %v", err)
+	} else if msr == nil {
+		utils.ErrExit("end migration: migration status record not found. Is the migration initialized?")
+	}
+	retrieveMigrationUUID()
+	checkIfEndCommandCanBePerformed(msr)
+
+	// backing up the state from the export directory
+	backupSchemaFilesFn()
+	backupDataFilesFn()
+	backupLogFilesFn()
+	saveMigrationReportsFn()
+
+	// cleaning only the migration state wherever and  whatever required
+	cleanupSourceDB(msr)
+	cleanupTargetDB(msr)
+	cleanupFallForwardDB(msr)
+	cleanupFallBackDB(msr)
+
+	cleanupExportDir()
+	utils.PrintAndLog("Migration ended successfully")
+}
+
+func backupSchemaFilesFn() {
+	if !bool(backupSchemaFiles) || !utils.FileOrFolderExists(filepath.Join(exportDir, "schema")) {
```

Comment:
Nit: factor out the `filepath.Join` calls by storing the result in `schemaDirName` local variable.

No need to type-cast the `BoolStr` types.